modules/nw-ingress-controller-nodeportservice-projects.adoc

```diff
@@ -0,0 +1,190 @@
+// Module included in the following assemblies:
+//
+// * networking/configuring_ingress_cluster_traffic/nw-configuring-ingress-controller-endpoint-publishing-strategy.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="nw-ingress-controller-nodeportservice-projects_{context}"]
+= Adding a single NodePort service to an Ingress Controller
+
+Instead of creating a `NodePort`-type `Service` for each project, you can create a custom Ingress Controller to use the `NodePortService` endpoint publishing strategy. To prevent port conflicts, consider this configuration for your Ingress Controller when you want to apply a set of routes, through Ingress sharding, to nodes that might already have a `HostNetwork` Ingress Controller.  
+
+Before you set a `NodePort`-type `Service` for each project, read the following considerations:
+
+* You must create a wildcard DNS record for the Nodeport Ingress Controller domain. A Nodeport Ingress Controller route can be reached from the address of a worker node. For more information about the required DNS records for routes, see "User-provisioned DNS requirements".
+* You must expose a route for your service and specify the `--hostname` argument for your custom Ingress Controller domain.
+* You must append the port that is assigned to the `NodePort`-type `Service` in the route so that you can access application pods. 
+
+.Prerequisites
+
+* You installed the {oc-first}.
+* Logged in as a user with `cluster-admin` privileges.
+* You created a wildcard DNS record.
+// https://docs.openshift.com/container-platform/4.16/networking/ingress-controller-dnsmgt.html (does not detail how to create the DNS)
+
+.Procedure
+
+. Create a custom resource (CR) file for the Ingress Controller:
++
+.Example of a CR file that defines information for the `IngressController` object
+[source,yaml]
+----
+apiVersion: v1
+items:
+- apiVersion: operator.openshift.io/v1
+  kind: IngressController
+  metadata:
+    name: <custom_name> <1>
+    namespace: openshift-ingress-operator
+  spec:
+    replicas: 1
+    domain: <domain> <2>
+    nodePlacement:
+      nodeSelector:
+        matchLabels:
+          <key>: <value> <3>
+    namespaceSelector:
+     matchLabels:
+       <key>: <value> <4>
+    endpointPublishingStrategy:
+      type: NodePortService 
+kind: List
+metadata:
+  resourceVersion: ""
+  selfLink: ""
+# ...
+----
```

Comment:
We just create one custom ingresscontroller here so better not use `kind: List`, this part can be updated as
```
apiVersion: operator.openshift.io/v1
kind: IngressController
metadata:
  name: <custom_name> <1>
  namespace: openshift-ingress-operator
spec:
  replicas: 1
  domain: <domain> <2>
  nodePlacement:
    nodeSelector:
      matchLabels:
        <key>: <value> <3>
  namespaceSelector:
    matchLabels:
      <key>: <value> <4>
  endpointPublishingStrategy:
    type: NodePortService
```